External/FEXCore/include/FEXCore/Core/Context.h

```diff
@@ -73,220 +73,215 @@ namespace FEXCore::Context {
+      FEX_DEFAULT_VISIBILITY virtual void DestroyContext() = 0;
+
+      /**
+       * @brief Allows setting up in memory code and other things prior to launchign code execution
+       *
+       * @param CTX The context that we created
+       * @param Loader The loader that will be doing all the code loading
+       *
+       * @return true if we loaded code
+       */
+      FEX_DEFAULT_VISIBILITY virtual FEXCore::Core::InternalThreadState* InitCore(uint64_t InitialRIP, uint64_t StackPointer) = 0;
+
+      FEX_DEFAULT_VISIBILITY virtual void SetExitHandler(ExitHandler handler) = 0;
+      FEX_DEFAULT_VISIBILITY virtual ExitHandler GetExitHandler() const = 0;
+
+      /**
+       * @brief Pauses execution on the CPU core
+       *
+       * Blocks until all threads have paused.
+       */
+      FEX_DEFAULT_VISIBILITY virtual void Pause() = 0;
+
+      /**
+       * @brief Starts (or continues) the CPU core
+       *
+       * This function is async and returns immediately.
+       * Use RunUntilExit() for synchonous executions
+       *
+       */
+      FEX_DEFAULT_VISIBILITY virtual void Run() = 0;
+
+      /**
+       * @brief Tells the core to shutdown
+       *
+       * Blocks until shutdown
+       */
+      FEX_DEFAULT_VISIBILITY virtual void Stop() = 0;
+
+      /**
+       * @brief Executes one instruction
+       *
+       * Returns once execution is complete.
+       */
+      FEX_DEFAULT_VISIBILITY virtual void Step() = 0;
+
+      /**
+       * @brief Runs the CPU core until it exits
+       *
+       * If an Exit handler has been registered, this function won't return until the core
+       * has shutdown.
+       *
+       * @param CTX The context that we created
+       *
+       * @return The ExitReason for the parentthread.
+       */
+      FEX_DEFAULT_VISIBILITY virtual ExitReason RunUntilExit() = 0;
+
+      FEX_DEFAULT_VISIBILITY virtual void CompileRIP(FEXCore::Core::InternalThreadState *Thread, uint64_t GuestRIP) = 0;
+
+      /**
+       * @brief Gets the program exit status
+       *
+       *
+       * @param CTX The context that we created
+       *
+       * @return The program exit status
+       */
+      FEX_DEFAULT_VISIBILITY virtual int GetProgramStatus() const = 0;
+
+      /**
+       * @brief [[threadsafe]] Returns the ExitReason of the parent thread. Typically used for async result status
+       *
+       * @param CTX The context that we created
+       *
+       * @return The ExitReason for the parentthread
+       */
+      FEX_DEFAULT_VISIBILITY virtual ExitReason GetExitReason() = 0;
+
+      /**
+       * @brief [[theadsafe]] Checks if the Context is either done working or paused(in the case of single stepping)
+       *
+       * Use this when the context is async running to determine if it is done
+       *
+       * @param CTX the context that we created
+       *
+       * @return true if the core is done or paused
+       */
+      FEX_DEFAULT_VISIBILITY virtual bool IsDone() = 0;
```

Comment:
Should this be const qualified?

Reply:
Yes. It wasn't previously const qualified. Switched it over to being const qualified.